### PR TITLE
Feature/create super class

### LINKED ISSUE #6: Multiple classes have methods with the exact same functionality.
- Each class should not have separate functions that do the same thing.  
- Maybe create a parent class with methods that can be utilized by multiple classes

### PR DESCRIPTION
**Pull request by jdxsmith**
#### What’s this PR do?
- Incorporates `inheritance` by creating a parent class and having the child classes extend the parent class
- Removes unnecessary functions from the child classes
- Rewrote tests to account for `inheritance`

#### Where should the reviewer start?
- Start in `Health-monitor.js` and see how the methods take the place of commented out methods in the child classes

#### How should this be manually tested?
- Run `npm test` to confirm all refactored tests are passing

#### Any background context you want to provide?
- Understand the importance of `inheritance` and why we used it

#### What are the relevant tickets?
- Fixes #6

### CODE REVIEW

**Comment by jaypeasee** [comment]
Made a lot of changes here:
1. We were missing tests for `calculateDaily` so I added two there on the `healthMonitor` class
1. Removed now redundant tests and methods that the `healthMonitor`
1. Removes all the commented out code from the PR presented.

I think this resolves redundancies, but it would be good to get eyes on this against the previous PR to make sure I didn't remove anything absolutely necessary.

**Comment by jaypeasee** [comment]
This Last push mainly just recycles the `calculateDaily()` method twice to DRY up some methods in the `Activity` class.

**Comment by jdxsmith** [comment]
> This Last push mainly just recycles the `calculateDaily()` method twice to DRY up some methods in the `Activity` class.

- I'm glad you were able to clean this up after we got the basic functionality down
- Good job catching that `calculateDaily()` can be applied in more places than we originally realized.  Changes like that definitely clean up our code.

File: src/Activity.js [hunk 1]
```diff
@@ -1,24 +1,17 @@
...
-    });
-    return parseFloat(((userStepsByDate.numSteps * userRepo.strideLength) / 5280).toFixed(1));
-  }
-
-  getActiveMinutesByDate(id, date) {
-    const userActivityByDate = this.activityData.find(data => {
-      return id === data.userID && date === data.date
-    });
-    return userActivityByDate.minutesActive;
+    const userStepsByDate = this.calculateDaily(id, date, "numSteps");
```
Comment (Codo-Baggins) [inline src/Activity.js:9]: Great job DRYing this up by \ calling the calculateDaily() method in here and inside of the accomplishStepGoal() method to get rid of code that was nearly identical.

**Review by Codo-Baggins — APPROVED** [review]

**Pull request merged**